agent_memory_server/cli.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python
+"""
+Command-line interface for agent-memory-server.
+"""
+
+import datetime
+import importlib
+import sys
+
+import click
+import uvicorn
+
+from agent_memory_server.config import settings
+from agent_memory_server.logging import configure_logging, get_logger
+from agent_memory_server.utils.redis import ensure_search_index_exists, get_redis_conn
+
+
+configure_logging()
+logger = get_logger(__name__)
+
+VERSION = "0.2.0"
+
+
+@click.group()
+def cli():
+    """Command-line interface for agent-memory-server."""
+    pass
+
+
+@cli.command()
+def version():
+    """Show the version of agent-memory-server."""
+    click.echo(f"agent-memory-server version {VERSION}")
+
+
+@cli.command()
+@click.option("--port", default=settings.port, help="Port to run the server on")
+@click.option("--host", default="0.0.0.0", help="Host to run the server on")
+@click.option("--reload", is_flag=True, help="Enable auto-reload")
+def api(port: int, host: str, reload: bool):
+    """Run the REST API server."""
+    import asyncio
+
+    from agent_memory_server.main import app, on_start_logger
+
+    async def setup_redis():
+        redis = await get_redis_conn()
+        await ensure_search_index_exists(redis)
+
+    # Run the async setup
+    asyncio.run(setup_redis())
+
+    on_start_logger(port)
+    uvicorn.run(
+        app,
+        host=host,
+        port=port,
+        reload=reload,
+    )
+
+
+@cli.command()
+@click.option("--port", default=settings.mcp_port, help="Port to run the MCP server on")
+@click.option("--sse", is_flag=True, help="Run the MCP server in SSE mode")
+def mcp(port: int, sse: bool):
+    """Run the MCP server."""
+    import asyncio
+
+    # Update the port in settings FIRST
+    settings.mcp_port = port
+
+    # Import mcp_app AFTER settings have been updated
+    from agent_memory_server.mcp import mcp_app
+
+    async def setup_and_run():
+        redis = await get_redis_conn()
+        await ensure_search_index_exists(redis)
+
+        # Run the MCP server
+        if sse:
+            await mcp_app.run_sse_async()
+        else:
+            await mcp_app.run_stdio_async()
+
+    # Update the port in settings
+    settings.mcp_port = port
+
+    click.echo(f"Starting MCP server on port {port}")
+
+    if sse:
+        click.echo("Running in SSE mode")
+    else:
+        click.echo("Running in stdio mode")
+
+    asyncio.run(setup_and_run())
+
+
+@cli.command()
+@click.argument("task_path")
+@click.option(
+    "--args",
+    "-a",
+    multiple=True,
+    help="Arguments to pass to the task in the format key=value",
+)
+def schedule_task(task_path: str, args: list[str]):
```

Comment:
Ooo I may very well steal this for Docket :D

Reply:
Hah! Nice! This was my MVP for scheduling recurring tasks. Cron, here we come!

Reply:
Oh!  Check out `Perpetual`, it's not documented that awesomely, but I think it's what you're looking for: https://github.com/chrisguidry/docket/blob/main/tests/test_fundamentals.py#L973-L997

Docket doesn't have cron syntax, but it can reschedule tasks on intervals forever

Reply:
Oh nice, I'll check that out!